test/runtests.jl

```diff
@@ -1204,6 +1204,8 @@ end
     @test_throws BoundsError ones(3)[Trues(2)]
     @test_throws BoundsError setindex!(ones(3), zeros(3), Trues(2))
     @test_throws DimensionMismatch setindex!(ones(2), zeros(3), Trues(2))
+    @test Ones(3)[Trues(3)] == Ones(3)
```

Comment:
Add test with offset axes

Reply:
With the restricted I added above, now something like
`Ones((2:4,))[Trues(3)]`
throws an error about method ambiguity, which seems appropriate.
I'm assuming we don't want a test for that.
The manual hints that there is an ambiguity for linear indexing
https://docs.julialang.org/en/v1/devdocs/offset-arrays/#Linear-indexing-(LinearIndices)
but does not seem to address the logical indexing case.

Reply:
That’s fine, didn’t realise that things were so ill-defined